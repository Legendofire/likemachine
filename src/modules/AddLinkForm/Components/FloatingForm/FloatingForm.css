.floatingform__container{
  box-sizing: border-box;
  display: block;
  height: 250px;
  width: 500px;
  border-radius: 20px;
  background-color: #8181E4;
  position: absolute;
  top: 100%;
  right: 30px;
  z-index: 9;
  padding-left: 30px;
  padding-right: 30px;
}

.appear{
  animation: appear 0.3s ease-in;
  animation-fill-mode: forwards;
}

.disappear{
  animation: disappear 0.3s ease-in;
  animation-fill-mode: forwards;
}

@keyframes appear {
  0% { top: 100%; }
  100% { top: calc(100% - 360px);}
}

@keyframes disappear {
  0% { top: calc(100% - 360px);}
  100% { top: 100%; }
}

.floatingform__container h2{
  color: white;
}

.form{
  display: flex;
  flex-direction: column;
}

.formGroup {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
}

.formGroup input{
  height: 30px;
  margin-bottom: 10px;
  width: 70%;
  border: none;
}

.formGroup label{
  height: 20px;
  margin-bottom: 10px;
  width: 20%;
  color: white;

}

.form button{
  width: 40%;
  height: 30px;
  border: none;
  background-color: white;
  align-self: flex-end;
}
