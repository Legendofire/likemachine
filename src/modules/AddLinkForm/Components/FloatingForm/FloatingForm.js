import React, { Component } from 'react';

import "./FloatingForm.css";

export default class FloatingForm extends Component{
  constructor(props){
    super(props);
    this.state = {
      "title": "",
      "url": "",
      "img": ""
    }
    this.updateFormData = this.updateFormData.bind(this);
  }

  updateFormData(target){
    this.setState({[target.name]:target.value});
  }

  render(){
    return (
      <div className={this.props.isFormOpen === true ? "floatingform__container appear":"floatingform__container disappear"}>
        <h2>Add New Link</h2>
        <div className="form">
          <div className="formGroup">
            <label>Title:</label>
            <input onChange={(e)=>this.updateFormData(e.target)} name="title" type="text"/>
          </div>
          <div className="formGroup">
            <label>URL:</label>
            <input onChange={(e)=>this.updateFormData(e.target)} name="url" type="text"/>
          </div>
          <div className="formGroup">
            <label>Image URL:</label>
            <input onChange={(e)=>this.updateFormData(e.target)} name="img" type="text"/>
          </div>
          <button onClick={()=>{this.props.formSubmitAction(this.state,this.props.session_id)}}>Add</button>
        </div>
      </div>
    )
  }
}
