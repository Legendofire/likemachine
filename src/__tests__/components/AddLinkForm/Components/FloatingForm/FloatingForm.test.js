import React from 'react';
import FloatingForm from '../../../../../modules/AddLinkForm/Components/FloatingForm/FloatingForm';
import renderer from 'react-test-renderer';

test('Floating Form disappears when open is false', () => {
  const component = renderer.create(
    <FloatingForm isFormOpen={false} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Floating Form disappears when open is true', () => {
  const component = renderer.create(
    <FloatingForm isFormOpen={true} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
