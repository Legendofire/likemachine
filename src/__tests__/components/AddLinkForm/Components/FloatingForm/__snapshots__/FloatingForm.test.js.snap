// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Floating Form disappears when open is false 1`] = `
<div
  className="floatingform__container disappear"
>
  <h2>
    Add New Link
  </h2>
  <div
    className="form"
  >
    <div
      className="formGroup"
    >
      <label>
        Title:
      </label>
      <input
        name="title"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="formGroup"
    >
      <label>
        URL:
      </label>
      <input
        name="url"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="formGroup"
    >
      <label>
        Image URL:
      </label>
      <input
        name="img"
        onChange={[Function]}
        type="text"
      />
    </div>
    <button
      onClick={[Function]}
    >
      Add
    </button>
  </div>
</div>
`;

exports[`Floating Form disappears when open is true 1`] = `
<div
  className="floatingform__container appear"
>
  <h2>
    Add New Link
  </h2>
  <div
    className="form"
  >
    <div
      className="formGroup"
    >
      <label>
        Title:
      </label>
      <input
        name="title"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="formGroup"
    >
      <label>
        URL:
      </label>
      <input
        name="url"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="formGroup"
    >
      <label>
        Image URL:
      </label>
      <input
        name="img"
        onChange={[Function]}
        type="text"
      />
    </div>
    <button
      onClick={[Function]}
    >
      Add
    </button>
  </div>
</div>
`;
